packages/text-input/src/TextInput.tsx

```diff
@@ -323,11 +323,7 @@ const TextInput: React.ComponentType<
       }
     }
 
-    if (!label && !ariaLabelledBy) {
-      console.error(
-        'For screen-reader accessibility, label or aria-labelledby must be provided to TextInput.',
-      );
-    }
+    validateLabelProps({ label, ...rest }, 'TextInput');
```

Comment:
Ditto here. Can you validate we're switching this to equivalent usages elsewhere as well?

Reply:
There are two functions, validateLabelProps and validateAriaLabelProps, this is because there's a difference in what's required based on the component's usage. For example, we're requiring that form inputs have a label... if a consumer does not want to use our styled label, then we need to verify that they have another label on the page which is why aria-label doesn't cut it, and we need an aria-labelledby. That's different from an implementation like SideNav, where we don't need an explicit label rendered on the screen, and therefore their options are aria-label or aria-labelledby

Reply:
Then I'd ask, does this need to be generalized? This seems like a very different use-case, and very component specific. It's not a check that there's a programmatically-determinable label, but that there is a visual label in the first place which is more of a design consideration. I don't think a utility function here is the right solution, and I'd rather keep this PR to its intent – moving validateAriaLabelProps to the A11y package.

If anything, we could make a component that handles the label and description rendering that we do in a few components. We can make a ticket to address this issue more specifically.